lib/galaxy/datatypes/registry.py

```diff
@@ -130,9 +131,9 @@ def __import_module(full_path: str, datatype_module: str):
             #           type="galaxy.datatypes.blast:BlastXml" />
             compressed_sniffers = {}
             handling_proprietary_datatypes = False
-            if isinstance(config, str):
+            if isinstance(config, (str, Path)):
                 # Parse datatypes_conf.xml
-                tree = galaxy.util.parse_xml(config)
+                tree = galaxy.util.parse_xml(str(config))
```

Comment:
Maybe we could push the `str()` casting to https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/util/__init__.py#L275 ?